CAN1-76 Top Level Requirements

djstedronsky opened the pull request:
Added top level requirements document - Devon

djstedronsky committed:
CAN1-76 Top Level Requirements

Added top level requirements document - Devon

PricelessAntonio reviewed (changes requested):
Need to address comments in the document below:
[top_level_requirements_comments_v1.docx](https://github.com/hruy1/CAN_we_do_it_Can_Controller/files/4940079/top_level_requirements_comments_v1.docx)
[Top_Level_Requirements_Checklist_v1.docx](https://github.com/hruy1/CAN_we_do_it_Can_Controller/files/4940080/Top_Level_Requirements_Checklist_v1.docx)

djstedronsky committed:
CAN1-76 Top Level Requirements Revision 2

Revision 2 - Devon

djstedronsky commented:
Revision notes: 
Added rising edge conditions where necessary and removed generic clocking requirements.
Read and write requirements are meant to say reserved and r/w only in the cases where a register is incompatible with the operation being attempted.
Interrupt requirements are unique because they are different methods of clearing the interrupt (by clearing the enable, or clearing the status bit)
The consecutive operation requirements for read and write are testable by keeping the cs high and showing that only one operation is completed. Toggle the cs signal and see if another operation is completed without having to toggle the RNW signal.

djstedronsky committed:
CAN1-76 Top level requirements revision 3

final revision top level requirements - Devon

PricelessAntonio commented:
Congrats, you passed!
[Top_Level_Requirements_Checklist_v2.docx](https://github.com/hruy1/CAN_we_do_it_Can_Controller/files/4940542/Top_Level_Requirements_Checklist_v2.docx)
